PWGHF/TableProducer/HFLbToLcPiCandidateSelector.cxx

```diff
@@ -0,0 +1,159 @@
+
+  Configurable<std::vector<double>> pTBins{"pTBins", std::vector<double>{hf_cuts_lb_tolcpi::pTBins_v}, "pT bin limits"};
+  Configurable<LabeledArray<double>> cuts{"Lb_to_lcpi_cuts", {hf_cuts_lb_tolcpi::cuts[0], npTBins, nCutVars, pTBinLabels, cutVarLabels}, "Lb0 candidate selection per pT bin"};
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc+"};
+
+  // Apply topological cuts as defined in HFSelectorCuts.h; return true if candidate passes all cuts
+  template <typename T1, typename T2, typename T3>
+  bool selectionTopol(const T1& hfCandLb, const T2& hfCandLc, const T3& trackPi)
+  {
+    auto candpT = hfCandLb.pt();
+    int pTBin = findBin(pTBins, candpT);
+    if (pTBin == -1) {
+      // Printf("Lb topol selection failed at getpTBin");
+      return false;
+    }
+
+    //pi pt
+    if (trackPi.pt() < cuts->get(pTBin, "pT Pi")) {
+      return false;
+    }
+
+    //d0(Lc)xd0(pi) ==> removed temporarily
+    if (hfCandLb.impactParameterProduct() > cuts->get(pTBin, "Imp. Par. Product")) {
+      return false;
+    }
+
+    //Lc mass ==> removed temporarily
+    //if (trackPi.sign() < 0) {
+    //if (std::abs(InvMassLcpKpi(hfCandLc) - RecoDecay::getMassPDG(pdg::Code::kLambdaCPlus)) > cuts->get(pTBin, "DeltaMLc")) {
+    //  return false;
+    //}
+    //}
+
+    //Lb Decay length
+    if (hfCandLb.decayLength() < cuts->get(pTBin, "Lb decLen")) {
+      return false;
+    }
+
+    //Lb Decay length XY
+    if (hfCandLb.decayLengthXY() < cuts->get(pTBin, "Lb decLenXY")) {
+      return false;
+    }
+
+    //Lb CPA cut
+    if (hfCandLb.cpa() < cuts->get(pTBin, "CPA")) {
+      return false;
+    }
+
+    //if (candpT < pTCandMin || candpT >= pTCandMax) {
+    // Printf("B+ topol selection failed at cand pT check");
+    // return false;
+    // }
+
+    //B+ mass cut
+    //if (std::abs(InvMassBPlus(hfCandBPlus) - RecoDecay::getMassPDG(521)) > cuts->get(pTBin, "m")) {
+    // Printf("B+ topol selection failed at mass diff check");
+    //  return false;
+    // }
+
+    //d0 of D0 and pi
+    //if ((std::abs(hfCandBPlus.impactParameter0()) > cuts->get(pTBin, "d0 D0")) ||
+    //    (std::abs(hfCandBPlus.impactParameter1()) > cuts->get(pTBin, "d0 Pi"))){
+    //  return false;
+    //}
+
+    //D0 CPA
+    // if (std::abs(hfCandD0.cpa()) < cuts->get(pTBin, "CPA D0")){
+    //  return false;
+    //}
+    return true;
+  }
+
+  void process(aod::HfCandLb const& hfCandLbs, soa::Join<aod::HfCandProng3, aod::HFSelLcCandidate>)
+  {
+    for (auto& hfCandLb : hfCandLbs) { //looping over Lb candidates
+
+      int statusLb = 0;
+
+      // check if flagged as Λb --> Λc+ π-
+      if (!(hfCandLb.hfflag() & 1 << hf_cand_lb::DecayType::LbToLcPi)) {
+        hfSelLbToLcPiCandidate(statusLb);
+        //Printf("Lb candidate selection failed at hfflag check");
+        continue;
+      }
+
+      // Lc is always index0 and pi is index1 by default
+      auto candLc = hfCandLb.index0();
```

Comment:
You have to cast it with `index0_as<...>` to the type of the table you get the Lc candidates from.